test/e2e/set/scylladbmonitoring/scylladbmonitoring.go

```diff
@@ -436,3 +497,357 @@ func decodeGrafanaDashboardFromGZBase64String(s string) (*grafanaDashboard, erro
+
+	framework.By("Creating a ClusterRoleBinding for external Prometheus")
+	_, err = f.KubeAdminClient().RbacV1().ClusterRoleBindings().Create(ctx, &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-prometheus", sm.Name),
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      prometheusServiceAccountName,
+				Namespace: f.Namespace(),
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     prometheusClusterRoleName,
+		},
+	}, metav1.CreateOptions{
+		FieldManager: f.FieldManager(),
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	return sa
+}
+
+// createServiceForPrometheus creates a headless Service for Prometheus to be used by external Prometheus.
+func createServiceForPrometheus(ctx context.Context, f *framework.Framework, sm *scyllav1alpha1.ScyllaDBMonitoring) *corev1.Service {
+	g.GinkgoHelper()
+
+	framework.By("Creating a Service for Prometheus")
+	prometheusName := prometheusNameForScyllaDBMonitoring(sm)
+	svc, err := f.KubeAdminClient().CoreV1().Services(f.Namespace()).Create(ctx, &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      prometheusName,
+			Namespace: f.Namespace(),
+		},
+		Spec: corev1.ServiceSpec{
+			ClusterIP: corev1.ClusterIPNone,
+			Selector: map[string]string{
+				"app.kubernetes.io/managed-by": "prometheus-operator",
+				"app.kubernetes.io/name":       "prometheus",
+				"app.kubernetes.io/instance":   prometheusName,
+				"operator.prometheus.io/name":  prometheusName,
+				"prometheus":                   prometheusName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name:     "web",
+					Port:     9090,
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+		},
+	}, metav1.CreateOptions{
+		FieldManager: f.FieldManager(),
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	return svc
+}
+
+func createExternalPrometheusInstanceWithoutTLS(ctx context.Context, f *framework.Framework, sm *scyllav1alpha1.ScyllaDBMonitoring, sa *corev1.ServiceAccount, svc *corev1.Service) *monitoringv1.Prometheus {
+	g.GinkgoHelper()
+
+	framework.By("Creating a Prometheus instance")
+	prom, err := f.PrometheusOperatorAdminClient().MonitoringV1().Prometheuses(f.Namespace()).Create(ctx, &monitoringv1.Prometheus{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      prometheusNameForScyllaDBMonitoring(sm),
+			Namespace: f.Namespace(),
+		},
+		Spec: monitoringv1.PrometheusSpec{
+			CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
+				ServiceAccountName: sa.Name,
+				ServiceName:        pointer.Ptr(svc.Name),
+				SecurityContext: &corev1.PodSecurityContext{
+					RunAsNonRoot: pointer.Ptr(true),
+					RunAsUser:    pointer.Ptr[int64](65534),
+					FSGroup:      pointer.Ptr[int64](65534),
+				},
+				ServiceMonitorSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"scylla-operator.scylladb.com/scylladbmonitoring-name": sm.Name,
```

Comment:
same here, I think the wiring would be much more understandable if the constructors were "scylladbmonitoring agnostic" - i.e. they took everything needed for wiring with SDBM as parameters, instead of creating it from ScyllaDBMonitoring the same way we do for managed instances